fix for MIPS architecture

NoUsername opened the pull request:
replaced all usb data parameters with kmalloc'ed memory

on certain MIPS architectures supplying stack memory as data argument to the usb functions is prohibited (the kernel needs this memory to be aligned correctly) otherwise a "kernel oops" happens and the system gets more or less unstable.

This fixes this issue. Tested it on a small embedded device (MIPS) and my laptop (x86).

NoUsername committed:
fix for MIPS architecture

replaced all usb data parameters with kmalloc'ed memory

scosu commented:
hi,

first of all thanks for the patch. I will add some line comments to the patch. Also please add the pull request description as commit message.

Also regarding kernel rules, your signed-off-by tag is missing in the commit.

Best Regards,

Markus

NoUsername commented:
hy, wow thank you for all the feedback, I'll try to work through it, really appreciate it!

One question so far: What do you mean with this comment: "Also regarding kernel rules, your signed-off-by tag is missing in the commit."
I'm afraid I'm not sure what that means / what i have to do - what "kernel rules" and what's a "signed-off-by tag"?

NoUsername committed:
Fix issues discussed in merge request

replaced all usb data parameters with kmalloc'ed memory

on certain MIPS architectures supplying stack memory as data argument to the usb functions is prohibited (the kernel needs this memory to be aligned correctly) otherwise a "kernel oops" happens and the system gets more or less unstable.

This fixes this issue. Tested it on a small embedded device (MIPS) and my laptop (x86).

scosu commented:
Hi,

The 'kernel rules' are documented in 'Documentation/SubmittingPatches'  
[1]. The document describes how to submit a patch to the Linux kernel.              

Chapter 12 of that document describes why all commits need a  
signed-off-by tag. As I plan to submit the egpms driver to the mainline  
kernel, I need to make sure that all submitted patches are GPL2. Also it  
is important to have the correct coding style and so on.                            

I hope that answers your question, if you have any further questions,  
please ask.                                                                         

Are you using an egpms power outlet with some MIPS router or something  
else?                                                                               

Best regards,                                                                       

Markus                                                                              

[1]  
https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/tree/Documentation/SubmittingPatches
